lte/gateway/c/session_manager/SessionState.h

```diff
@@ -267,6 +267,8 @@ class SessionState {
 
   std::string get_imsi() const { return config_.common_context.sid().id(); }
 
+  uint16_t get_shard_id() { return shard_id_; }
```

Comment:
add set_shard_id as int  the template and use that function to modify the value, no constructors

The reason to use set is the same I mentioned beforeL: it is way better to have just one single entry point to see where the property is modified. Also if we ever do rebalancing we will need to use set_shard_id anyway